lib/mix/lib/mix/tasks/deps.compile.ex

```diff
@@ -189,13 +189,34 @@ defmodule Mix.Tasks.Deps.Compile do
     lib_path = Path.join(config[:env_path], "lib/*/ebin")
 
     env = [{"REBAR_CONFIG", config_path}, {"TERM", "dumb"}]
-    cmd = "#{rebar_cmd(dep)} bare compile --paths=#{inspect(lib_path)}"
+
+    cmd = rebar_compile_cmd(dep, dep_path, lib_path)
 
     File.mkdir_p!(dep_path)
     File.write!(config_path, rebar_config(dep))
     do_command(dep, config, cmd, false, env)
   end
 
+  defp rebar_compile_cmd(dep, build_path, lib_path) do
+    rebar_path = rebar_cmd(dep)
+    vsn = rebar_vsn(rebar_path)
+    rebar_compile_cmd_for(vsn, rebar_path, inspect(build_path), inspect(lib_path))
+  end
+
+  defp rebar_compile_cmd_for(nil, rebar_path, _build_path, lib_path) do
+    "#{rebar_path} bare compile --paths #{lib_path}"
+  end
+
+  defp rebar_compile_cmd_for(vsn, rebar_path, build_path, lib_path) do
+    case Version.compare("3.12.2", vsn) do
```

Comment:
3.13.2 is the first one to support the output directory and the first one to be annotated.

Reply:
Yeah, I wondered if doing the Version comparison was overkill... sounds like it.

Reply:
You don't have to do a comparison, you just have to check if it is exists.